app/assets/javascripts/listings/ListingUnitService.js.coffee

```diff
@@ -129,13 +213,53 @@ ListingUnitService = ($http, ListingConstantsService, ListingIdentityService) ->
     combined = Service.combineUnitSummaries(listing)
     _.every(combined, { Unit_Type: 'SRO' })
 
+
+  Service.getListingAMI = (listing) ->
+    angular.copy([], Service.AMICharts)
+    Service.loading.ami = true
+    Service.error.ami = false
+    # shouldn't happen, but safe to have a guard clause
+    return $q.when() unless listing.chartTypes
+    allChartTypes = _.sortBy(listing.chartTypes, 'percent')
+    console.log('listing chart types', listing.chartTypes, 'allChartTypes', allChartTypes)
+
+    data =
+      'year[]': _.map(allChartTypes, 'year')
+      'chartType[]': _.map(allChartTypes, 'chartType')
+      'percent[]': _.map(allChartTypes, 'percent')
+    $http.get('/api/v1/listings/ami.json', { params: data }).success((data, status, headers, config) ->
+      if data && data.ami
+        angular.copy(Service._consolidatedAMICharts(data.ami), Service.AMICharts)
+      Service.loading.ami = false
```

Comment:
do you need to reset error.ami here? (only if this will ever be called on a retry or something)

Reply:
looks like we reset it at the top of the function instead 
https://github.com/SFDigitalServices/sf-dahlia-web/pull/1319/files/39cf8c5f15c903c823947e86518c13822fe19c38#diff-c23158f12ed2799cdb7b127595de12dbR220